platform/configurator/configurator-api/src/main/java/org/codice/ddf/admin/configurator/OperationReport.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator;
+
+import java.util.List;
+
+/**
+ * <b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public interface OperationReport {
+    boolean txactSucceeded();
+
+    Result getResult(String key);
+
+    List<Result> getFailedResults();
```

Comment:
My concern here is that a user of this interface should expect one thing and not something based on the implementation. I have another comment below that actually suggest to change it to a stream instead of a list. But if you keep it as a list, I think we would want to indicate what the user can and cannot do with the list.